app/src/androidTest/java/com/google/jetpackcamera/utils/UiTestUtil.kt

```diff
@@ -192,15 +195,60 @@ fun deleteFilesInDirAfterTimestamp(
     return hasDeletedFile
 }
 
-fun doesImageFileExist(uri: Uri, prefix: String): Boolean {
-    val file = uri.path?.let { File(it) }
-    if (file?.exists() == true) {
-        val mimeType = URLConnection.guessContentTypeFromName(uri.path)
-        return mimeType != null && mimeType.startsWith(prefix)
+fun doesFileExist(uri: Uri): Boolean = uri.path?.let { File(it) }?.exists() == true
+
+fun doesMediaExist(uri: Uri, prefix: String): Boolean {
+    if (prefix == IMAGE_PREFIX) {
+        return doesImageExist(uri)
+    } else if (prefix == VIDEO_PREFIX) {
+        return doesVideoExist(uri, prefix)
     }
     return false
 }
 
+fun doesImageExist(uri: Uri): Boolean {
+    val bitmap = uri.path?.let { path -> BitmapFactory.decodeFile(path) }
+    val mimeType = URLConnection.guessContentTypeFromName(uri.path)
+    return mimeType != null && mimeType.startsWith(IMAGE_PREFIX) && bitmap != null
+}
+
+// this does not work with the image capture tests...
+fun doesVideoExist(
+    uri: Uri,
+    prefix: String,
+    checkAudio: Boolean = false,
+    durationMs: Long? = null
+): Boolean {
+    if (!doesFileExist(uri)) {
+        return false
+    }
+    var hasCorrectContents = MediaMetadataRetriever().useAndRelease {
+        try {
+            it.setDataSource(uri.path)
+        } catch (e: Exception) {
+            Log.e(TAG, e.stackTraceToString())
+            throw e
+            return@useAndRelease false
+        }
+        if (!it.getMimeType().startsWith(prefix)) {
+            return@useAndRelease false
+        }
+        if (prefix == VIDEO_PREFIX && !it.hasVideo()) {
+            return@useAndRelease false
+        }
+        if (checkAudio && !it.hasAudio()) {
+            return@useAndRelease false
+        }
+
+        if (durationMs != null && it.getDurationMs() != durationMs) {
+            return@useAndRelease false
+        }
+
+        return@useAndRelease true
+    }
+    return hasCorrectContents == true
```

Comment:
Assuming you check that `VIDEO_PREFIX` is used for the prefix and you get rid of the `try/catch` block, this set of cases / return statements can be simplified to:
```
    return MediaMetadataRetriever().useAndRelease {
        it.setDataSource(uri.path)

        it.getMimeType().startsWith(prefix) &&
        it.hasVideo() &&
        (!checkAudio || it.hasAudio()) &&
        (durationMs == null || it.getDurationMs() == durationMs)
    } == true
```